### PR TITLE
manifest: hal_espressif update

### PR DESCRIPTION
**Pull request by marekmatej**
Update the hal_espressif to include latest esptool version.
Improve support for new SoCs as such as ESP32-C6.
Fix bugs related to segment alignment.

Fix issue related in PR #73437

### CODE REVIEW

**Review by str4t0m — DISMISSED** [review]
Please don't use the commit sha of the unmerged hal PR. Use `pull/208/head` until it's merged.

**Review by sylvioalves — APPROVED** [review]

**Comment by marekmatej** [comment]
> Please don't use the commit sha of the unmerged hal PR. Use `pull/208/head` until it's merged.

hi @str4t0m thanks for the comment. I used `ref` as related PR was on the verge of merging. Can you revisit this, please?

**Comment by str4t0m** [comment]
My change request has been addressed and no longer blocks the PR.
With a second approval it may be merged.
I would have to test and have a closer look at the changes before giving an approval which I won't be able to do before evening.

**Review by LucasTambor — APPROVED** [review]

**Pull request merged**
